crates/integrations/datafusion/src/table.rs

```diff
@@ -82,6 +82,26 @@ impl TableProvider for IcebergTableProvider {
         Ok(Arc::new(IcebergTableScan::new(
             self.table.clone(),
             self.schema.clone(),
+            _filters,
         )))
     }
+
+    fn supports_filters_pushdown(
+        &self,
+        filters: &[&Expr],
+    ) -> std::result::Result<Vec<TableProviderFilterPushDown>, datafusion::error::DataFusionError>
+    {
+        let filter_support = filters
+            .iter()
+            .map(|e| {
+                if let Expr::BinaryExpr(BinaryExpr { .. }) = e {
```

Comment:
using match to better support other Expr?

Reply:
sure.